src/Source.h

```diff
@@ -9,47 +11,93 @@ typedef boost::shared_ptr<Source> SourcePtr;
 
 class Source {
 public:
+  Source() {}
   virtual ~Source() {}
 
   virtual const char* begin() = 0;
   virtual const char* end() = 0;
+  inline std::string encoding() {
+    if (pEncoding == "") {
+      stop("Encoding not set at Source");
+    } else {
+      return pEncoding;
+    }
+  }
+  inline const std::vector<std::string>& get_comments() { return pComments; }
+
+  void set_comments(std::vector<std::string> utf8comments) {
```

Comment:
The `set_comments` is used by the SourceString.h, SourceRaw.h and SourceFile.h classes. It takes care of storing the comments and, if an encoding was given, convert them to that encoding. I can replace the `set_comments` call with an assignment and `reencode_comments()` and remove that method if you want.

The Source class had the `begin()` and `end()` virtual methods that return a `const char *begin_` or `end_` in all derivative classes. Having something like `comments()` (the `get_` part maybe is unnecessary) or `encoding()` makes sense to me for consistence.